sw/device/lib/testing/test_framework/ottf_console.c

```diff
@@ -324,3 +364,13 @@ size_t ottf_console_spi_device_read(size_t buf_size, uint8_t *const buf) {
 
   return received_data_len;
 }
+
+status_t ottf_console_putbuf(void *io, const char *buf, size_t len) {
+  size_t written_len = sink(io, buf, len);
+  if (len != written_len) {
+    return DATA_LOSS((int32_t)written_len);
```

Comment:
wouldn't the data lost be `len - written_len`?

Reply:
Good point! My original thought was to report the actual data written length along with a data loss error, but it seems to be reasonable to report the length of missing data.